embassy-nrf/src/util.rs

```diff
@@ -19,10 +19,11 @@ pub(crate) fn slice_in_ram<T>(slice: *const [T]) -> bool {
     ptr >= SRAM_LOWER && (ptr + len * core::mem::size_of::<T>()) < SRAM_UPPER
 }
 
-/// Return an error if slice is not in RAM.
+/// Return an error if slice is not in RAM. Skips check if slice is zero-length.
 #[cfg(not(feature = "nrf51"))]
 pub(crate) fn slice_in_ram_or<T, E>(slice: *const [T], err: E) -> Result<(), E> {
-    if slice_in_ram(slice) {
+    let (_, len) = slice_ptr_parts(slice);
+    if len > 0 && slice_in_ram(slice) {
```

Comment:
This will always fail if len==0, we want it to always succeed.

Reply:
Good catch. I apparently shouldn't author PRs during the middle of the night 🙈

This should do the trick:
```suggestion
    if len == 0 || slice_in_ram(slice) {
```